src/Service/Payment.php

```diff
@@ -443,4 +472,28 @@ public static function getAllowed(array $customFields): array
             Payment::REFUND_AMOUNT => $customFields[Form::CUSTOM_FIELD_WORLDLINE_PAYMENT_TRANSACTION_REFUND_AMOUNT] / 100,
         ];
     }
+
+
+    /**
+     * @param string $currentStatus
+     * @param string $goalStatus
+     * @return bool
+     */
+    public static function operationImossible(string $currentStatus, string $goalStatus): bool
+    {
+        if ($currentStatus === $goalStatus) {
+            return true;
+        }
+
+        if (!array_key_exists($currentStatus, Payment::POSSIBLE_STATUSES)) {
```

Comment:
is statuses really a word?
I think the plural for status is status or just use states :)